src/api/v4/layer/layer.js

```diff
@@ -376,6 +423,13 @@ Layer.prototype._createInternalModel = function (engine) {
 
 // Internal functions.
 
+Layer.prototype.$setClient = function (client) {
+  if (this._client) {
```

Comment:
Shouldn't we check if `client` is the same instance as we do with `engine`? Just in case.

Reply:
It should be always the same client for the engine. But just is case

Reply:
Ok. If this comes from `engine` and the check is done at that level we're good to go then.